zndraw/static/pycom/Bookmarks.js

```diff
@@ -0,0 +1,66 @@
+class Bookmarks {
+  constructor(world, cache, socket) {
+    this.world = world;
+    this.cache = cache;
+    this.socket = socket;
+    this.bookmarks = {};
+
+    this.socket.on(
+      "bookmarks:get",
+      function (callback) {
+        console.log("bookmarks:get");
+        callback(this.bookmarks);
+      }.bind(this),
+    );
+    this.socket.on("bookmarks:set", (bookmarks) => {
+      console.log("bookmarks:set");
+      this.bookmarks = bookmarks;
+      this.updateBookmarks();
+    });
+
+    // on keypress b set the bookmark
+    window.addEventListener("keypress", (event) => {
```

Comment:
Add shift-B to cycle through created bookmarks?

Reply:
Also, how to delete bookmars?

Reply:
I've added shift + arrow left/right to iterate and shift + click the bookmark to remove it.